DOM Manipulation and Events: Switch order of Assignments #29305

FranVlahovic opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
Switched order of assignments in order to finish JavaScript Tutorial's reading before diving into MDN Active Learning Section to test student skills.


## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->

- Moved MDN Active learning section below JavaScript tutorial reading.

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Closes #29305 

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

ManonLef commented:
@FranVlahovic I notice this lesson has some [linting issues](https://github.com/TheOdinProject/curriculum/pull/30052/files) unrelated to your change. Let me know if you would like to address these as well in this PR. If not, I will open an issue for them.

FranVlahovic commented:
I will do all of it today, as soon as possible.

FranVlahovic commented:
> > > @FranVlahovic thank you and take your time, no need to rush :)
> > 
> > 
> > You can see if its alright now, i think i fixed blank line issues of my PR and i added headings to fix "Note boxes must start with level 4 heading". Although in my opinion its better if its done without headings, since in few of linting issues i've put "_Tip_" as a heading but i think its not really appropriate maybe "_Note_" would work..
> 
> As per the layout style guide, note box headings should briefly describe the note box contents and not just be something generic like "Note" or "Tip". This is for accessibility so that they can be navigated to via meaningful headings, which also means they cannot be repeated (a link to the `Note` heading will never link to the second one instance).

Alright i will fix it accordingly today.

ManonLef commented on foundations/javascript_basics/DOM_manipulation_and_events.md:
```diff
@@ -220,6 +224,8 @@ Keep in mind that the JavaScript does *not* alter your HTML, but the DOM - your
 
 <div id="important-note" class="lesson-note" markdown="1">
 
+#### Important
```
```suggestion
#### Timing of JavaScript
```

I think we should make this more descriptive

ManonLef commented on foundations/javascript_basics/DOM_manipulation_and_events.md:
```diff
@@ -68,6 +68,8 @@ When your HTML code is parsed by a web browser, it is converted to the DOM, as w
 
 <div class="lesson-note">
 
+#### Performance Consideration
```
```suggestion
#### Performance consideration
```

ManonLef commented on foundations/javascript_basics/DOM_manipulation_and_events.md:
```diff
@@ -178,7 +180,9 @@ div.innerHTML = "<span>Hello World!</span>";
 <div class="lesson-note lesson-note--tip" markdown="1">
 
-Note that using textContent is preferred over innerHTML for adding text, as innerHTML should be used sparingly to avoid potential security risks.
+#### Security Risks of innerHTML
```
```suggestion
#### Security risks of innerHTML
```

ManonLef reviewed (changes requested):
Thank you for editing the headers. Some fixes are still required. You can change these locally, or click the "commit suggestion" buttons

FranVlahovic commented:
Now it should be fine, thanks for suggestions and for opportunity.

ManonLef commented:
@FranVlahovic I should thank you! It's appreciated
